src/smc-webapp/project_files.cjsx

```diff
@@ -1987,21 +1987,27 @@ exports.ProjectFiles = rclass ({name}) ->
                 style   = {error_style}
                 onClose = {=>@props.actions.setState(error:'')} />
 
+    render_public_view_not_logged_in_error_message: ->
+        <div>You are not logged in. To view non-public files or edit files in this project you{"'"}ll need to <Button bsStyle={'success'} onClick={=>$('.sign_in *').trigger('click')}>sign in</Button>. If you are not a collaborator then you need to ask a collaborator of the project to add you to access non public files.</div>
```

Comment:
Going to need a much better reason to let this in. Definitely do this a different way.